src/setup/testnet/mod.rs

```diff
@@ -0,0 +1,205 @@
+//! Utilities for setting up a testnet consisting of 3 nodes.
+
+use std::{
+    fmt,
+    fmt::Write,
+    fs, io,
+    net::{IpAddr, SocketAddr},
+    path::{Path, PathBuf},
+};
+
+use crate::setup::{
+    config::{DEFAULT_PORT, ZIGGURAT_CONFIG, ZIGGURAT_DIR},
+    node::{Node, NodeBuilder},
+};
+
+/// Testnet's directory for nodes' configs.
+const TESTNET_DIR: &str = "testnet";
+
+/// Validators file name.
+const VALIDATORS_FILE_NAME: &str = "validators.txt";
+
+const VALIDATOR_KEYS: [&str; 3] = [
+    "nHUSqn9qjEF7JJkVqvY7BFLMKdqP5KLLEjo5oB4QH43ADDndRawB",
+    "nHUEsvSFTf1Snr7ZUdLxjcMW6PKcMrwwXCGZBg6xb1ePG8R4C3TS",
+    "nHUuYdS49cPfRmCXPTwu7MVVFZFFmfG7y5sRttirVMhwuD7xStQp",
+];
+
+/// A struct allowing for convenient starting and stopping a small testnet.
+pub struct TestNet {
+    // Setup information for each node. Used for writing configuration.
+    setups: [NodeSetup; 3],
+    // Running nodes. Used to stopping the testnet.
+    running: Vec<Node>,
+    // Sets whether to log the node's output to Ziggurat's output stream.
+    use_stdout: bool,
+    // Path under which all nodes will be build
+    path: PathBuf,
+}
+
+impl TestNet {
+    #[allow(dead_code)] // for some bizarre reason clippy --all-targets reports this function as unused :-/
```

Comment:
I think it might be because it is only ever called in the test below _and_ it is private? If you make it public, clippy should no longer complain?

Reply:
Done